mmdet/apis/ote/apis/detection/base_task.py

```diff
@@ -0,0 +1,352 @@
+        logger.info(f"Scratch space created at {self._scratch_space}")
+
+        self._hyperparams = hyperparams = task_environment.get_hyper_parameters(OTEDetectionConfig)
+
+        self._model_name = hyperparams.algo_backend.model_name
+        self._labels = task_environment.get_labels(False)
+
+        template_file_path = task_environment.model_template.model_template_path
+
+        # Get and prepare mmdet config.
+        self._base_dir = os.path.abspath(os.path.dirname(template_file_path))
+        config_file_path = os.path.join(self._base_dir, hyperparams.algo_backend.model)
+        self._config = Config.fromfile(config_file_path)
+        patch_config(self._config, self._scratch_space, self._labels, random_seed=42)
+        set_hyperparams(self._config, hyperparams)
+
+        # Extra control variables.
+        self._training_work_dir = None
+        self._is_training = False
+        self._should_stop = False
+
+    @staticmethod
+    def _create_model(config: Config, from_scratch: bool = False):
+        """
+        Creates a model, based on the configuration in config
+
+        :param config: mmdetection configuration from which the model has to be built
+        :param from_scratch: bool, if True does not load any weights
+
+        :return model: ModelEntity in training mode
+        """
+        model_cfg = copy.deepcopy(config.model)
+
+        init_from = None if from_scratch else config.get('load_from', None)
+        logger.warning(init_from)
+        if init_from is not None:
+            # No need to initialize backbone separately, if all weights are provided.
+            model_cfg.pretrained = None
+            logger.warning('build detector')
+            model = build_detector(model_cfg)
+            # Load all weights.
+            logger.warning('load checkpoint')
+            load_checkpoint(model, init_from, map_location='cpu')
+        else:
+            logger.warning('build detector')
+            model = build_detector(model_cfg)
+        return model
+
+
+    def infer(self, dataset: Dataset, inference_parameters: Optional[InferenceParameters] = None) -> Dataset:
+        """ Analyzes a dataset using the latest inference model. """
+        set_hyperparams(self._config, self._hyperparams)
+
+        if inference_parameters is not None:
+            update_progress_callback = inference_parameters.update_progress
+            is_evaluation = inference_parameters.is_evaluation
+        else:
+            is_evaluation = False
+            update_progress_callback = default_progress_callback
+
+        time_monitor = InferenceProgressCallback(len(dataset), update_progress_callback)
+
+        def pre_hook(module, input):
+            time_monitor.on_test_batch_begin(None, None)
+
+        def hook(module, input, output):
+            time_monitor.on_test_batch_end(None, None)
+
+        pre_hook_handle = self._model.register_forward_pre_hook(pre_hook)

```

Comment:
What's the reason to make this `OTEBaseTask` at all? It's neither abstract, nor functional. Another thing is that it's hard to track changes that are made to the current implementation of `OTEDetectionTask`. Why don't you directly inherit `NNCFDetectionTask` from `OTEDetectionTask`?

Reply:
Main difference between `NNCFDetectionTask` and `OTEDetectionTask` is that one inherited from `IOptimizationTask` another from `ITrainingTask`. `NNCFDetectionTask` should not have interface from `ITrainingTask` and can not be inherited from `OTEDetectionTask`. As result moving common functions to base class looks logically.